sources/packages/api/src/services/atbc/atbc.service.ts

```diff
@@ -0,0 +1,180 @@
+import { Injectable } from "@nestjs/common";
+import { ConfigService } from "../config/config.service";
+import axios from "axios";
+import { LoggerService } from "../../logger/logger.service";
+import { InjectLogger } from "../../common";
+import {
+  ATBCHeader,
+  ATBCAuthTokenResponse,
+} from "../auth/token-cache.service.models";
+import {
+  ATBCIntegrationConfig,
+  ATBCCreateClientResponse,
+  ATBCCreateClientPayload,
+  ATBCPDCheckerResponse,
+  ATBCPDCheckerPayload,
+} from "../../types";
+import { Student } from "../../database/entities";
+import { StudentService } from "../../services";
+
+@Injectable()
+export class ATBCService {
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly studentService: StudentService,
+  ) {
+    this.logger.log("[Created]");
+  }
+  /**
+   * Allow a singleton access to the ATBCService for areas
+   * of the application that are not part of the dependency
+   * injection framework.
+   */
+  public static get shared(): ATBCService {
```

Comment:
With a few exceptions, services created to be used alongside Nestjs do not need to have singletons created. Please remove it and ensure that it is being injected.
By default, the services injected by Nest will be singletons unless that we do not want it and change its default behavior using decorators.
More info: https://docs.nestjs.com/fundamentals/injection-scopes